sound/soc/sof/intel/hda.c

```diff
@@ -417,11 +428,21 @@ static irqreturn_t hda_dsp_interrupt_thread(int irq, void *context)
 {
 	struct snd_sof_dev *sdev = context;
 
-	if (hda_dsp_check_ipc_irq(sdev))
-		sof_ops(sdev)->irq_thread(irq, sdev);
-	if (hda_dsp_check_stream_irq(sdev))
+	/* deal with streams and controller first */
+	if (sdev->irq_event & SOF_HDA_IRQ_STREAM ||
+	    hda_dsp_check_stream_irq(sdev))
```

Comment:
@plbossart @bardliao I was late to review as this is already merged, but isn't the "irq_event" flag superfluous here when we anyways call the check_foobar_irq() function in the thread?

Reply:
it's a minor optimization, if we know the source needs to be dealt with we don't recheck in the thread. we can remove this if no one thinks it's useful, I didn't share the code upstream yet to allow for additional fixes/improvements.

The other point I am not sure about is if we need to use snd_sof_dsp_update_bits() or snd_sof_dsp_update_bits_unlocked() for the GIE changes. it does look like I am using the same hw_lock twice.

Reply:
@kv2019i @plbossart Can we check Global Interrupt Status (GIS) in the handler if we don't use irq_event? Also, we may not need to lock sdev->hw_lock in hda_dsp_check_ipc_irq() any more. I will test it today.

Reply:
> @kv2019i @plbossart Can we check Global Interrupt Status (GIS) in the handler if we don't use irq_event? Also, we may not need to lock sdev->hw_lock in hda_dsp_check_ipc_irq() any more. I will test it today.

sorry, can you elaborate, I didn't get your question @bardliao

Reply:
> > @kv2019i @plbossart Can we check Global Interrupt Status (GIS) in the handler if we don't use irq_event? Also, we may not need to lock sdev->hw_lock in hda_dsp_check_ipc_irq() any more. I will test it today.
> 
> sorry, can you elaborate, I didn't get your question @bardliao

@plbossart Please check https://github.com/thesofproject/linux/pull/1554